DSCResources/MSFT_xRemoteFile/MSFT_xRemoteFile.psm1

```diff
@@ -252,8 +252,15 @@ function Test-TargetResource
                     Write-Debug "Cache is empty or it doesn't reflect current state. File will be downloaded."
                 }
             } else {
-                Write-Debug "MatchSource is false. No need for downloading file."
-                $fileExists = $true 
+                Write-Debug "Destination file size is not the same as source file. File will be downloaded."
+                [System.Net.WebClient]$wc = [System.Net.WebClient]::New()
+                $wc.OpenRead($URI)
+                [Long]$bytes_total = $wc.ResponseHeaders["Content-Length"]
+                if($bytes_total -eq (Get-Item $DestinationPath).Length)
+                {
+                    Write-Debug "MatchSource is false. No need for downloading file."

```

Comment:
I'd also modify this message to make it more accurate, something like:
"Size of the destination and source files match. No need for downloading the file"